@stellar/design-system/src/components/Asset/index.tsx

```diff
@@ -0,0 +1,76 @@
+import "./styles.scss";
+
+/** */
+export type AssetSource = {
+  /** Image URL */
+  image: string;
+  /** Image alt text */
+  altText: string;
+  /** Custom background color */
+  backgroundColor?: string;
+};
+
+/** */
+export type AssetBaseProps = {
+  /** Asset size */
+  size: "sm" | "md" | "lg";
+  /** First asset source */
+  sourceOne: AssetSource;
+};
+
+/** */
+export type SingleAssetProps = {
+  /** Asset or asset pair variant */
+  variant: "single";
+  sourceTwo?: undefined;
+};
+
+/** */
+export type MultiAssetProps = {
+  /** Asset or asset pair variant */
+  variant: "swap" | "pair" | "platform";
+  /** Second asset source */
+  sourceTwo: AssetSource;
+};
+
+/** */
+export type AssetProps = (SingleAssetProps | MultiAssetProps) & AssetBaseProps;
```

Comment:
I experimented with using discriminated unions to have a better developer experience when using props that can't be used together. For example, a `single` asset can't have `sourceTwo` prop. This makes prop rendering on the website a bit more complicated, but overall it's a better experience when using the component.